_posts/2017-10-12-Rust-1.21.md

```diff
@@ -0,0 +1,210 @@
+first steps toward having them work on stable Rust. Please check out the
+preview, and you'll hear more about these plans in the future.
+
+Finally, a few documentation improvements. First up, if you visit [the docs
+for `std::os`] (https://doc.rust-lang.org/stable/std/os/), which contains
+operating-system specific functionality, you'll now see more than just `linux`,
+the platform we build the documentation on. We've long regretted that the hosted
+version of the documentation has been Linux-specific; this is a first step towards
+rectifying that. This is [specific to the standard
+library](https://github.com/rust-lang/rust/pull/43348) and not for general use;
+we'll hope to improve this further in the future.
+
+Next, [Cargo's docs are moving!](https://github.com/rust-lang/rust/pull/43916)
+Historically, Cargo's docs were hosted on doc.crates.io, which doesn't follow
+the release train model, even though Cargo itself does. This lead to situations
+where a feature would land in Cargo nightly, the docs would be updated, and
+then for up to twelve weeks, users would *think* that it should work, but it
+wouldn't yet. [https://doc.rust-lang.org/cargo](https://doc.rust-lang.org/cargo)
+will be the new home of Cargo's docs, though for now, that URL is a redirect to
+doc.crates.io. Future releases will move Cargo's docs over, and at that point,
+doc.crates.io will redirect to doc.rust-lang.org/cargo. Cargo's docs have long
+needed a refreshing, so expect to hear more news about Cargo's docs generally
+through the end of the year.
+
+Finally, until now, `rustdoc` did not have any documentation. This is now
+[fixed](https://github.com/rust-lang/rust/pull/43863), with a new "`rustdoc`
+Book," located at
+[https://doc.rust-lang.org/rustdoc](https://doc.rust-lang.org/rustdoc). These
+docs are fairly bare-bones at the moment, but we'll be improving them over
+time.
+
+See the [detailed release notes][notes] for more.
+
+#### Library stabilizations
+
+Not too many stabilizations this release, but there's one really great
+quality of life change: due to the lack of type-level integers, arrays only
+supported various traits up to size 32. This [has now been fixed for the
+`Clone` trait](https://github.com/rust-lang/rust/pull/43690), which also
+caused a lot of ICEs at times, when a type would be `Copy` but not `Clone`.
+For other traits, [an RFC for type-level integers was accepted
+recently](https://github.com/rust-lang/rfcs/blob/master/text/2000-const-generics.md),
+which may help with this situation. That change has yet to be implemented, however,
+though pre-requisite work is ongoing at the moment.
+
+[`Rc<T>` and `Arc<T>` now implement `From<&[T]> where T: Clone`, `From<str>`,
+`From<String>`, `From<Box<T>> where T: ?Sized`, and
+`From<Vec<T>>`.](https://github.com/rust-lang/rust/pull/42565)
+
+Finally, [`std::mem::discriminant` has been
+stabilized](https://doc.rust-lang.org/std/mem/fn.discriminant.html), allowing
+you to see what variant an `enum` is.
```

Comment:
Maybe add "without using matching"? This may be misleading to newcomers otherwise.

Reply:
And maybe "an `enum` value" or "an `enum` instance`"?